File: articles/linear-algebra-for-systems-modeling/articles/simulation-of-high-dimensional-systems/calculators/r/high_dimensional_simulation_calculator.R
result <- data.frame(
  calculator = "simulation_of_high_dimensional_systems_calculator",
  model_name = "synthetic_high_dimensional_simulation_audit",
  state_dimension = 24,
  time_steps = 40,
  ensemble_runs = 250,
  method = "sparse_linear_state_update_with_correlated_monte_carlo_shocks",
  random_seed = 20260629,
  transition_spectral_radius = 0.94,
  transition_density = 0.12,
  final_state_mean_norm = 4.8,
  final_state_mean_total = 24.6,
  final_state_95th_percentile_total = 26.0,
  threshold_exceedance_probability = 0.10,
  first_three_component_energy = 0.78,
  warning = "Simulation metrics depend on state representation, transition rules, uncertainty assumptions, covariance, random seed, ensemble size, dimensionality reduction, validation, and interpretation context."
)

dir.create("outputs", recursive = TRUE, showWarnings = FALSE)
write.csv(result, "outputs/r_simulation_of_high_dimensional_systems_calculator.csv", row.names = FALSE)
print(result)

File: articles/linear-algebra-for-systems-modeling/articles/simulation-of-high-dimensional-systems/calculators/python/high_dimensional_simulation_calculator.py
from __future__ import annotations

import csv
import json
from pathlib import Path


def main() -> None:
    output_dir = Path("outputs")
    output_dir.mkdir(parents=True, exist_ok=True)

    result = {
        "calculator": "simulation_of_high_dimensional_systems_calculator",
        "model_name": "synthetic_high_dimensional_simulation_audit",
        "state_dimension": 24,
        "time_steps": 40,
        "ensemble_runs": 250,
        "method": "sparse_linear_state_update_with_correlated_monte_carlo_shocks",
        "random_seed": 20260629,
        "transition_spectral_radius": 0.94,
        "transition_density": 0.12,
        "final_state_mean_norm": 4.8,
        "final_state_mean_total": 24.6,
        "final_state_95th_percentile_total": 26.0,
        "threshold_exceedance_probability": 0.10,
        "first_three_component_energy": 0.78,
        "warning": "Simulation metrics depend on state representation, transition rules, uncertainty assumptions, covariance, random seed, ensemble size, dimensionality reduction, validation, and interpretation context."
    }

    with (output_dir / "simulation_of_high_dimensional_systems_calculator.json").open("w", encoding="utf-8") as handle:
        json.dump(result, handle, indent=2, sort_keys=True)

    with (output_dir / "simulation_of_high_dimensional_systems_calculator.csv").open("w", newline="", encoding="utf-8") as handle:
        writer = csv.DictWriter(handle, fieldnames=list(result.keys()))
        writer.writeheader()
        writer.writerow(result)

    print(json.dumps(result, indent=2, sort_keys=True))


if __name__ == "__main__":
    main()
